lib/Github/HttpClient/Builder.php

```diff
@@ -181,6 +182,9 @@ public function addHeaderValue($header, $headerValue)
      */
     public function addCache(CacheItemPoolInterface $cachePool, array $config = [])
     {
+        if (!isset($config['cache_key_generator'])) {
+            $config['cache_key_generator'] = new HeaderCacheKeyGenerator(['Authorization', 'Cookie', 'Accept']);
```

Comment:
I think this will be sufficient! Maybe @GrahamCampbell can test this pr to see if it fixes his problem from #557